AxonDeepSeg/train_network.py

```diff
@@ -54,588 +58,178 @@ def train_model(path_trainingset, path_model, config, path_model_init=None,
 
-                    with open(path_model / 'evolution_stats.pkl', 'wb') as handle:
-                        pickle.dump(evolution, handle)
+    # Image and Mask Data Generator
+    image_generator_train = image_datagen.flow(image_train_gen, y=None, seed=seed, batch_size=batch_size, shuffle=True)
+    mask_generator_train = mask_datagen.flow(mask_train_gen, y=None, seed=seed, batch_size=batch_size, shuffle=True)
 
-                    file = open(path_model / "report.txt", 'w')
-                    file.write(Report + output_2)
-                    file.close()
+    # Just zip the two generators to get a generator that provides augmented images and masks at the same time
+    train_generator = zip(image_generator_train, mask_generator_train)
 
-                # Increase the epoch #
-                epoch += 1
+    ###########################Validation###########
+    data_gen_args = dict()
+    image_datagen = ImageDataGenerator(**data_gen_args)
+    mask_datagen = ImageDataGenerator(**data_gen_args)
 
-            # Increase the step #
-            step += 1
+    seed = 2018
 
-        # At the end of the training we save the model in a checkpoint file
-        save_path = saver.save(session, str(path_model / "model.ckpt"))
+    image_generator_valid = image_datagen.flow(x=image_valid_gen, y=None, seed=seed, batch_size=batch_size,
+                                               shuffle=False)
+    mask_generator_valid = mask_datagen.flow(x=mask_valid_gen, y=None, seed=seed, batch_size=batch_size, shuffle=False)
 
-        # Initialize best model with model after epoch 1
-        evolution = {'loss': Loss, 'steps': Epoch, 'accuracy': Accuracy}
-        with open(path_model / 'evolution.pkl', 'wb') as handle:
-            pickle.dump(evolution, handle)
+    # Just zip the two generators to get a generator that provides augmented images and masks at the same time
+    valid_generator = zip(image_generator_valid, mask_generator_valid)
 
-        print(("Model saved in file: %s" % save_path))
-        print("Optimization Finished!")
+    ########################### Initalizing U-Net Model ###########
+    model = uconv_net(config, bn_updated_decay=None, verbose=True)
 
+    ########################### Tensorboard for Visualization ###########
+    # Name = "SEM_3c_dataset-{}".format(time.strftime("%Y-%m-%d %H:%M:%S", time.gmtime()))
+    tensorboard = TensorBoard(log_dir=str(path_model))
 
+    ########################## Training Unet Model ###########
 
-def pw_dices(prediction, gt):
-    """
-    Computes the pixel-wise dice from the prediction tensor outputted by the network.
-    :param prediction: Tensor, the prediction outputted by the network. Shape (N,H,W,C).
-    :param gt: Tensor, the gold standard we work with. Shape (N,H,W,C).
-    :return: Vector, dice per class for the current batch.
-    """
+    # Adam Optimizer for Unet
+    adam = keras.optimizers.Adam(lr=learning_rate, beta_1=0.9, beta_2=0.999, epsilon=1e-08, decay=0.0)
```

Comment:
Can we have an explanation of how the hardcoded values for the Adam Optimizer were chosen?

Reply:
These are the default values in Keras Adam Optimizer. Since these are default values so I had removed them in commit (7f1712c).